SwiftCheck/Modifiers.swift

```diff
@@ -239,15 +276,15 @@ public struct ArrowOf<T : protocol<Hashable, CoArbitrary>, U : Arbitrary> : Arbi
 		self.arr = arr
 	}
 
-	public init(_ arr : (T -> U)) {
+	public convenience init(_ arr : (T -> U)) {
 		self.init(Dictionary(), { (_ : T) -> U in return undefined() })
 
-		self.arr = { x in
-			if let v = self.table[x] {
+		self.arr = { [weak self] x in
+			if let v = self!.table[x] {
```

Comment:
Self is guaranteed to be alive in that call.  The ? is not any safer, so I should probably be using unretained or something.